tests/lava/proc/conv/test_models.py

```diff
@@ -0,0 +1,304 @@
+    """
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        shape = kwargs.get('shape', (1,))
+        buffer = kwargs.get('buffer')
+        self.shape = shape
+        self.a_in = InPort(shape=shape)
+        buffer_shape = shape + (buffer,)
+        self.data = Var(shape=buffer_shape, init=np.zeros(buffer_shape))
+
+
+class AbstractPyReceiveModel(PyLoihiProcessModel):
+    """Template receive process model."""
+    def run_spk(self):
+        """Receive spikes and store in an internal variable"""
+        data = self.a_in.recv()
+        buffer = self.data.shape[-1]
+        if verbose:
+            print(f'Received data={data[data!=0]}')
+        self.data[..., (self.current_ts - 1) % buffer] = data
+
+
+@implements(proc=ReceiveProcess, protocol=LoihiProtocol)
+@requires(CPU)
+@tag('floating_pt')
+class PyReceiveModelFloat(AbstractPyReceiveModel):
+    """Float receive process model."""
+    a_in: PyInPort = LavaPyType(PyInPort.VEC_DENSE, np.int32, precision=24)
+    data: np.ndarray = LavaPyType(np.ndarray, np.int32, precision=24)
+
+
+@implements(proc=ReceiveProcess, protocol=LoihiProtocol)
+@requires(CPU)
+@tag('fixed_pt')
+class PyReceiveModelFixed(AbstractPyReceiveModel):
+    """Fixed point receive process model."""
+    a_in: PyInPort = LavaPyType(PyInPort.VEC_DENSE, float)
+    data: np.ndarray = LavaPyType(np.ndarray, float)
+
+
+def setup_conv():
+    """Sets up random convolution setting."""
+    # conv parameter setup
+    groups = np.random.randint(4) + 1
+    in_channels = (np.random.randint(8) + 1) * groups
+    out_channels = (np.random.randint(8) + 1) * groups
+    kernel_size = np.random.randint([9, 9]) + 1
+    stride = np.random.randint([5, 5]) + 1
+    padding = np.random.randint([5, 5])
+    dilation = np.random.randint([4, 4]) + 1
+    weight_dims = [
+        out_channels,
+        kernel_size[0], kernel_size[1],
+        in_channels // groups
+    ]
+    weight = np.random.randint(256, size=weight_dims) - 128
+
+    # input needs to be a certain size
+    # to make sure the output dimension is never negative
+    input_shape = tuple(
+        (np.random.randint([128, 128]) + kernel_size * dilation).tolist()
+        + [in_channels]
+    )
+    output_shape = utils.output_shape(
+        input_shape, out_channels, kernel_size, stride, padding, dilation
+    )
+
+    conv = Conv(
+        input_shape=input_shape,
+        weight=weight,
+        stride=stride,
+        padding=padding,
+        dilation=dilation,
+        groups=groups,
+    )
+
+    params = {
+        'weight': weight,
+        'kernel_size': kernel_size,
+        'stride': stride,
+        'padding': padding,
+        'dilation': dilation,
+        'groups': groups,
+    }
+
+    return conv, input_shape, output_shape, params
+
+
+class TestConvProcessModels(unittest.TestCase):
+    """Tests for all ProcessModels of Conv"""
+
+    def test_source_sink(self):
+        """Test for source-sink process."""
```

Comment:
More details please: What does this test check? What is its purpose?

If necessary, provide inline comments to explain the intent of what you are doing.